### PR TITLE
Add cache size metrics to guave and caffeine

### PR DESCRIPTION
**Pull request by checketts**
Addresses #211

### CODE REVIEW

**Comment by checketts** [comment]
@stuartnelson3 I had to increase the MetricsFilter delta to get the tests to pass:

`java.lang.AssertionError: expected:<0.1> but was:<0.104399255>`

It was only allowing 0.001 difference. I let it allow 0.01. Let me know if that is too lenient and if there is a different, preferred approach.

File: simpleclient_servlet/src/test/java/io/prometheus/client/filter/MetricsFilterTest.java [hunk 1]
```diff
@@ -117,7 +117,7 @@ public Void answer(InvocationOnMock invocationOnMock) throws Throwable {
 
         final Double sum = CollectorRegistry.defaultRegistry.getSampleValue("foobar_baz_filter_duration_seconds_sum", new String[]{"path", "method"}, new String[]{path, HttpMethods.POST});
         assertNotNull(sum);
-        assertEquals(0.1, sum, 0.001);
+        assertEquals(0.1, sum, 0.01);
```
Comment (brian-brazil) [inline simpleclient_servlet/src/test/java/io/prometheus/client/filter/MetricsFilterTest.java]: Can you remove the unrelated change?

Comment (checketts) [inline simpleclient_servlet/src/test/java/io/prometheus/client/filter/MetricsFilterTest.java]: Without this change the test was flaky and failed for me locally and failed in TravisCI based on system load.

I'll remove it from the PR, but I suspect it is needed in the long term.

**Comment by checketts** [comment]
OK it is removed now.

**Comment by brian-brazil** [comment]
Thanks!
